File: ireader/src/main/java/com/book/ireader/model/bean/Source.java
package com.book.ireader.model.bean;

/**
 * Created with author.
 * Description:
 * Date: 2018-08-03
 * Time: 下午4:17
 */
public interface Source {
    public Class<?> getSourceClass();

    public String getSourceName();

    public String getSourceBaseUrl();

    public String getSourceSearchUrl();

    public String getSourceEncode();
}

File: app/src/main/java/com/book/novel/model/Source.kt
package com.book.novel.model

/**
 * Created with author.
 * Description:
 * Date: 2018/7/25
 * Time: 1:39
 */
interface Source {
    fun getSourceClass(): Class<Any>
    fun getSourceName(): String
    fun getSourceUrl(): String
}